src/main/java/com/google/api/codegen/transformer/py/PythonImportSectionTransformer.java

```diff
@@ -0,0 +1,179 @@
+  @Override
+  public ImportSectionView generateFileHeaderImportSection(SurfaceTransformerContext context) {
+    // TODO(eoogbe): implement this method when migrating to MVVM
+    return ImportSectionView.newBuilder().build();
+  }
+
+  @Override
+  public ImportSectionView generateInitCodeImportSection(
+      MethodTransformerContext context, Iterable<InitCodeNode> specItemNodes) {
+    return ImportSectionView.newBuilder()
+        .appImports(generateInitCodeAppImports(context, specItemNodes))
+        .build();
+  }
+
+  private List<ImportFileView> generateInitCodeAppImports(
+      MethodTransformerContext context, Iterable<InitCodeNode> specItemNodes) {
+    return ImmutableList.<ImportFileView>builder()
+        .add(generateApiImport(context))
+        .addAll(generateProtoImports(context, specItemNodes))
+        .addAll(generatePageStreamingImports(context))
+        .build();
+  }
+
+  private ImportFileView generateApiImport(MethodTransformerContext context) {
+    return createImport(
+        context.getApiConfig().getPackageName(),
+        context.getNamer().getApiWrapperVariableName(context.getInterface()));
+  }
+
+  private List<ImportFileView> generateProtoImports(
+      MethodTransformerContext context, Iterable<InitCodeNode> specItemNodes) {
+    ModelTypeTable typeTable = context.getTypeTable();
+    Set<ImportFileView> protoImports = new TreeSet<>(importFileViewComparator());
+    for (InitCodeNode item : specItemNodes) {
+      TypeRef type = item.getType();
+      if (!type.isRepeated() && type.isMessage()) {
+        String fullName = typeTable.getFullNameFor(type);
+        String nickname = typeTable.getNicknameFor(type);
+        protoImports.add(generateAppImport(fullName, nickname));
+      } else if (type.isEnum()) {
+        protoImports.add(createImport(context.getApiConfig().getPackageName(), "enums"));
+      }
+    }
+    return new ArrayList<>(protoImports);
+  }
+
+  private List<ImportFileView> generatePageStreamingImports(MethodTransformerContext context) {
+    ImmutableList.Builder<ImportFileView> pageStreamingImports = ImmutableList.builder();
+    if (context.getMethodConfig().isPageStreaming()) {
+      ImportTypeView callOptionsImport =
+          ImportTypeView.newBuilder().fullName("CallOptions").nickname("").build();
+      ImportTypeView initialPageImport =
+          ImportTypeView.newBuilder().fullName("INITIAL_PAGE").nickname("").build();
+      ImportFileView fileImport =
+          ImportFileView.newBuilder()
+              .moduleName("google.gax")
+              .types(ImmutableList.of(callOptionsImport, initialPageImport))
+              .build();
+      pageStreamingImports.add(fileImport);
+    }
+    return pageStreamingImports.build();
+  }
+
+  /**
+   * Orders the imports by: (1) number of attributes (least to most), (2) module name (A-Z), (3)
+   * attribute name (A-Z)
+   */
+  private Comparator<ImportFileView> importFileViewComparator() {
```

Comment:
`types` doesn't refer to std/external/app. The import section already divides the imports by those categories, though the init code section only has app imports. `types` refers to `ImportTypeView`s. Those contain a fullName (attributeName in python) and a nickname (localName in python). This comparator follows the lexicographic sort from `PythonImportHandler`, but doesn't convert everything to a string first.